docs/src/description/constraints.md

```diff
@@ -49,11 +49,11 @@ Boundary constraints can access public input values and random values provided b
 
 To use public inputs, the public input must be declared in the `public_inputs` source section. They can be accessed using array indexing syntax, as described by the [accessor syntax rules](./syntax.md#section-specific-accessors).
 
-Random values can be accessed by using array indexing syntax on the `$rand` built-in, as described by the [accessor syntax rules](./syntax.md#section-specific-accessors).
+To use random values, they must be declared in the `random_values` source section. Random values can be named using a single identifier for all the random values (e.g. rand: [16]) or using names for each random value or group of random values (e.g. `rand: [a, b[14], c]`). They can be accessed using `$` followed by the identifier name and the index of the random value (e.g. `$rand[10]`) or using the name of the named random value or group (e.g. `a` or `b[5]`), as described by the [accessor syntax rules](./syntax.md#section-specific-accessors).
```

Comment:
I think we can clarify this by using the bindings terminology here, since we've done that elsewhere, e.g.

> To use random values, they must be declared in the `random_values` source section. Random values are defined by an identifier which is bound to a fixed-length array (e.g. rand: [16]), but it is additionally possible to bind identifiers to specific random values or groups of values (e.g. `rand: [a, b[14], c]`). Random values can be accessed by using `$` followed by the array identifier and the index of the value (e.g. `$rand[10]`) or by using the name of the bound random value or group along with the index of the value within the group (e.g. `a` or `b[5]`), as described by the [accessor syntax rules](./syntax.md#section-specific-accessors).